includes/WikiDiscover.php

```diff
@@ -117,111 +107,149 @@ public function isInactive( $database ) {
 	public function isPrivate( $database ) {
 		return in_array( $database, $this->private );
 	}
+	
+	/**
+	 * @param Parser $parser
+	 */
+	public static function onParserFirstCallInit( Parser $parser ) {
+		$parser->setFunctionHook( 'numberofwikisincategory', [ __CLASS__, 'numberOfWikisInCategory' ], Parser::SFH_NO_HASH );
+		$parser->setFunctionHook( 'numberofwikisinlanguage', [ __CLASS__, 'numberOfWikisInLanguage' ], Parser::SFH_NO_HASH );
+		$parser->setFunctionHook( 'numberofwikisbysetting', [ __CLASS__, 'numberOfWikisBySetting' ], Parser::SFH_NO_HASH );
+	}
+	
+	/**
+	 * @param Parser $parser
+	 * @param string $category|null
+	 * @return integer
+	 */
+	public static function numberOfWikisInCategory( Parser $parser, String $category = null ) {
+		$config = MediaWikiServices::getInstance()->getMainConfig();
+
+		$dbw = wfGetDB( DB_MASTER, [], $config->get( 'CreateWikiDatabase' ) );
+
+		return $dbw->selectRowCount( 'cw_wikis', '*', [ 'wiki_deleted' => 0, 'wiki_category' => strtolower( $category ) ] );;
+	}
+
+	/**
+	 * @param Parser $parser
+	 * @param string $language|null
+	 * @return integer
+	 */
+	public static function numberOfWikisInLanguage( Parser $parser, String $language = null ) {
+		$config = MediaWikiServices::getInstance()->getMainConfig();
+
+		$dbw = wfGetDB( DB_MASTER, [], $config->get( 'CreateWikiDatabase' ) );
+
+		return $dbw->selectRowCount( 'cw_wikis', '*', [ 'wiki_deleted' => 0, 'wiki_language' => strtolower( $language ) ] );;
+	}
+	
+	/**
+	 * @param Parser $parser
+	 * @param mixed $setting|null
+	 * @param mixed $value|null
+	 * @return string|integer
+	 */
+	public static function numberOfWikisBySetting( $parser, $setting = null, $value = null ) {
+		if ( !$setting && !$value ) {
+			return 'Error: no input specified.';
+		}
+
+		$config = MediaWikiServices::getInstance()->getMainConfig();
+		$dbr = wfGetDB( DB_REPLICA, [], $config->get( 'CreateWikiDatabase' ) );
 
+		$extList = array_keys( $config->get( 'ManageWikiExtensions') );
+        
+		if ( !$value && !in_array( $setting, $extList ) ) {
+			return 0;
+		}
+        
+		if ( in_array( $setting, $extList ) ) {
+			$s_extensions = implode( ',', $dbr->selectFieldValues( 'mw_settings', 's_extensions' ) );
+			
+			$extensionUsageCount = substr_count( $s_extensions, '"' . $setting . '"' );
+
+			return $extensionUsageCount;
+		}
+
+		$s_settings = $dbr->selectFieldValues( 'mw_settings', 's_settings' );
+		$settingUsageCount = 0;
+
+		foreach( array_flip( $s_settings ) as $key ) {
```

Comment:
Why are you array flipping an array of strings?

Reply:
because the only thing that seemed to work was that or using `$key => $value` so was trying to avoid the unused `$value`

Reply:
It's an array of strings for me, it shouldn't be necessary to array_flip at all. I ran this portion of code on test2 and it worked without an array_flip.

Reply:
Oh, I must have changed something since then to make it work then. Will fix, thanks!

Reply:
done